### PR TITLE
DOCS: Documentation Style Guide

### PR DESCRIPTION
**Pull request by cafferata**
This PR replaces https://github.com/StackExchange/dnscontrol/pull/2373

_<https://docs.dnscontrol.org/~/revisions/OxJh5H8Ezyne7H42Ukua/developer-info/styleguide-doc>_

### CODE REVIEW

**Comment by cafferata** [comment]
My feedback has been incorporated and as far as I'm concerned this PR is good enough for a first version. 🚀

File: documentation/styleguide-doc.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+Within the git repo, docs are grouped:
+
+* [`documentation/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation): general docs
+* [`documentation/providers/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation/providers/): One file per provider
+* [`documentation/functions/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation/functions/): One file per `dnsconfig.js` language feature
+* [`documentation/assets/gcloud/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation/assets/gcloud): Images for page Google Cloud DNS (PNGs only, please!)
+
+## How to add a new page?
+
+1. Add the page to the `documentation` (or a sub folder)
```
Comment (tlimoncelli) [inline documentation/styleguide-doc.md:31]:
```suggestion
1. Add the page to the `documentation` folder (possibly a sub folder)
```

File: documentation/styleguide-doc.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+* **Step 2: Bar**
+
+Description of the step.
+
+(further sub sub headings are discouraged encouraged)
+```
+
+### Code Snippets
+
+> "Currently, we don't support adding code blocks inside other code blocks. We don't have plans to support that functionality at this time." — _GitBook Support (2023-03-29)_
```
Comment (tlimoncelli) [inline documentation/styleguide-doc.md:77]: The goal is to give the user something they can easily understand for including code with and without a filename.  Shorter examples are probably better (just 2-3 lines of code).  Maybe use a PNG to show what it looks like but link to something they can copy and paste?

File: documentation/styleguide-doc.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+   * Changes frequently.  Updated via [GitBook](https://www.gitbook.com/)
+
+**The directory structure**
+
+Within the git repo, docs are grouped:
+
+* [`documentation/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation): general docs
+* [`documentation/providers/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation/providers/): One file per provider
+* [`documentation/functions/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation/functions/): One file per `dnsconfig.js` language feature
+* [`documentation/assets/gcloud/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation/assets/gcloud): Images for page Google Cloud DNS (PNGs only, please!)
```
Comment (tlimoncelli) [inline documentation/styleguide-doc.md:27]: Please make this more generic.  For example...

```
* `documentation/assets/FOO/`: Images for page FOO (PNGs only, please)
```

**Comment by cafferata** [comment]
Rebased on master.

File: documentation/styleguide-doc.md (hunk 3 above)
Comment (cafferata) [reply documentation/styleguide-doc.md]: Fixed by https://github.com/StackExchange/dnscontrol/pull/2383/commits/2c8efa4001c86eeb7cf6f05cc5d1cc41f04be752. @tlimoncelli would you like to close this conversation after reviewing it?

File: documentation/styleguide-doc.md (hunk 2 above)
Comment (cafferata) [reply documentation/styleguide-doc.md]: I agree with you! I think removing the GitBook notes is enough. There are two examples, each with a deep link to the GitHub source.

Comment (cafferata) [reply documentation/styleguide-doc.md]: Fixed by https://github.com/StackExchange/dnscontrol/commit/9a65c77a2d1d1925d927aab45341bf5266dbd0c7. @tlimoncelli would you like to close this conversation after reviewing it?

**Comment by cafferata** [comment]
Feedback has been processed.

_<https://docs.dnscontrol.org/~/revisions/BRtPCFXEcJg89ztpW8cK/developer-info/styleguide-doc>_

File: documentation/styleguide-doc.md (hunk 1 above)
Comment (cafferata) [reply documentation/styleguide-doc.md]: _<https://docs.dnscontrol.org/~/revisions/BRtPCFXEcJg89ztpW8cK/developer-info/styleguide-doc#how-to-add-a-new-page>_

File: documentation/styleguide-doc.md (hunk 2 above)
Comment (cafferata) [reply documentation/styleguide-doc.md]: _<https://docs.dnscontrol.org/~/revisions/BRtPCFXEcJg89ztpW8cK/developer-info/styleguide-doc#code-snippets>_

File: documentation/styleguide-doc.md (hunk 3 above)
Comment (cafferata) [reply documentation/styleguide-doc.md]: _<https://docs.dnscontrol.org/~/revisions/BRtPCFXEcJg89ztpW8cK/developer-info/styleguide-doc#where-are-the-docs>_

**Comment by tlimoncelli** [comment]
Another example to include:

How to include a warning:

```
{% hint style="warning" %}
WARNING: Blah blah blah.
{% endhint %}
```

**Review by tlimoncelli — APPROVED** [review]
Looks good!

**Comment by cafferata** [comment]
> Looks good!

Thanks!



> Another example to include:
> 
> How to include a warning:

Yes, I'll work on it tomorrow.

**Comment by cafferata** [comment]
See the additional commits for feedback handling.
_<https://docs.dnscontrol.org/~/revisions/57p1yTu8hw59Lr4C5mX2/developer-info/styleguide-doc>_

**Comment by cafferata** [comment]
See the additional commits for feedback handling.
_<https://docs.dnscontrol.org/~/revisions/cW1o2Ylgeld8si24sIUp/developer-info/styleguide-doc>_

File: documentation/styleguide-doc.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+**Warning hints** are good for showing important information or non-critical warnings.
+{% endhint %}
+
+ [source code](markdown-examples/hint/hint-warning.md?plain=1)
+
+{% hint style="danger" %}
+**Danger hints** are good for highlighting destructive actions or raising attention to critical information.
+{% endhint %}
+
+ [source code](markdown-examples/hint/hint-dange.md?plain=1)
```
Comment (cafferata) [inline documentation/styleguide-doc.md:135]:
```suggestion
 [source code](markdown-examples/hint/hint-danger.md?plain=1)
```

**Comment by cafferata** [comment]
See the additional commits for feedback handling.
_<https://docs.dnscontrol.org/~/revisions/NeXurZYuMIinrW7pBMS2/developer-info/styleguide-doc>_

**Comment by cafferata** [comment]
See the additional commits for feedback handling.
_<https://docs.dnscontrol.org/~/revisions/AI4ZjTQL8AcwOcRcc8Bj/developer-info/styleguide-doc>_

**Review by tlimoncelli — APPROVED** [review]

**Comment by tlimoncelli** [comment]
Great! Thanks for filling in all the blanks!

**Pull request merged**